File: fundamentalsAlgo/Python/two_sum.py




def two_sum(nums,target):
    obj={}

    for i, val in enumerate(nums):
        diff=target-val
        if diff in obj:
            return (obj[diff], i)
        obj[val]=i
        

nums=[9,5,4,3]
target=12
print('The two sum is ', two_sum(nums, target))

File: fundamentalsAlgo/JavaScript/two_sum.js
function twoSum(nums, target) {
  const seen = new Map();

  for (let i = 0; i < nums.length; i += 1) {
    const diff = target - nums[i];
    if (seen.has(diff)) {
      return [seen.get(diff), i];
    }
    seen.set(nums[i], i);
  }

  return null;
}

const nums = [9, 5, 4, 3];
const target = 12;
console.log("The two sum is", twoSum(nums, target));

module.exports = { twoSum };
